src/ext/oblivc/obliv_common.h

```diff
@@ -126,24 +126,32 @@ static inline void setBit(char *dest,int ind,bool v)
 { char mask = (1<<ind%8);
   dest[ind/8] = (dest[ind/8]&~mask)+(v?mask:0);
 }
-static inline bool getBit(const char* src,int ind) 
+static inline bool getBit(const char* src,int ind)
   { return src[ind/8]&(1<<ind%8); }
-static inline void xorBit(char *dest,int ind,bool v) 
+static inline void xorBit(char *dest,int ind,bool v)
   { dest[ind/8]^=(v<<ind%8); }
 static inline void memxor (void* dest, const void* src, size_t n)
-{ size_t i;
-  for(i=0;i+sizeof(uint64_t)<=n;i+=sizeof(uint64_t))
-    *((uint64_t*)((char*)dest+i)) ^= *((uint64_t*)((char*)src+i));
-  if(i+sizeof(uint32_t)<=n){
-    *((uint32_t*)((char*)dest+i)) ^= *((uint32_t*)((char*)src+i));
-   i+=sizeof(uint32_t);
+{
+  // xor the last n % 8 bytes; Compiler optimization will unroll this loop
+  size_t offset = n % sizeof(uint64_t);
+  for(size_t i = n - offset; i < n; i++) {
+    ((char *)dest)[i] ^= ((char *)src)[i];
   }
-  if(i+sizeof(uint16_t)<=n)  {
-    *((uint16_t*)((char*)dest+i)) ^= *((uint16_t*)((char*)src+i));
-   i+=sizeof(uint16_t);
+  n /= sizeof(uint64_t);
```

Comment:
I realize that I am asking for microoptimizations here, but does it make any performance difference to start this wide xor from a 16-byte boundary?